File: features/payment/payment/src/main/java/com/tokopedia/payment/fingerprint/domain/FingerprintRepository.java
package com.tokopedia.payment.fingerprint.domain;

import com.tokopedia.payment.fingerprint.data.model.ResponsePaymentFingerprint;

import java.util.HashMap;

import rx.Observable;

/**
 * Created by zulfikarrahman on 3/23/18.
 */

public interface FingerprintRepository {
    Observable<Boolean> saveFingerprint(HashMap<String, Object> params);

    Observable<ResponsePaymentFingerprint> paymentWithFingerPrint(HashMap<String, Object> params);

    Observable<Boolean> savePublicKey(HashMap<String, String> params);

    Observable<HashMap<String, String>> getPostDataOtp(String transactionId);
}

File: features/payment/payment/src/main/java/com/tokopedia/payment/fingerprint/domain/FingerprintRepository.kt
package com.tokopedia.payment.fingerprint.domain

import com.tokopedia.payment.fingerprint.data.model.ResponsePaymentFingerprint
import rx.Observable
import java.util.*

/**
 * Created by zulfikarrahman on 3/23/18.
 */
interface FingerprintRepository {
    fun saveFingerprint(params: HashMap<String, Any?>): Observable<Boolean>
    fun paymentWithFingerPrint(params: HashMap<String, Any?>): Observable<ResponsePaymentFingerprint?>
    fun savePublicKey(params: HashMap<String, String?>): Observable<Boolean>
    fun getPostDataOtp(transactionId: String): Observable<HashMap<String, String>?>
}